### PR TITLE
allows for users to enter multiple beverages on order page

### LINKED ISSUE #76: Allow for users to enter multiple beverages comma separated

### PR DESCRIPTION
**Pull request by TylerHernandez**

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/RecommendServlet.java [hunk 1]
```diff
@@ -37,10 +37,25 @@
...
    */
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String beverage = request.getParameter("beverageRequested");
-    System.out.println(beverage);
-    List<String> listStoreIds = new ArrayList<>();
+    response.setContentType("application/json");
+    String userInputBeverages = request.getParameter("beverageRequested");
+    String[] beverages = userInputBeverages.split(",");
+    HashMap<String, String> beverageToStoreId = new HashMap<String, String>();
+    List<String> listStoreIds = new ArrayList<>(); // todo: get this list from the places api
```
Comment (mutisog) [inline src/main/java/com/google/sps/servlets/RecommendServlet.java:44]: nit: storeIds is sufficient.

File: src/main/webapp/order.html [hunk 2]
```diff
@@ -28,7 +28,7 @@
 
     <div> Enter Beverage: </div>
     <form>
-    <textarea id= "beverageRequested"> </textarea>
+    <textarea id= "beverageRequested"></textarea>
```
Comment (mutisog) [inline src/main/webapp/order.html:31]: Given the time constraints, could we manually test this functionality, and attach some screenshots to this PR?

**Review by dag13 — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/RecommendServlet.java [hunk 3]
```diff
@@ -37,10 +37,25 @@
    * the user specified beverage from (via a weighting formula)
    */
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String beverage = request.getParameter("beverageRequested");
-    System.out.println(beverage);
-    List<String> listStoreIds = new ArrayList<>();
+    response.setContentType("application/json");
+    String userInputBeverages = request.getParameter("beverageRequested");
+    String[] beverages = userInputBeverages.split(",");
+    HashMap<String, String> beverageToStoreId = new HashMap<String, String>();
```
Comment (albert-g3) [inline src/main/java/com/google/sps/servlets/RecommendServlet.java:43]: right side can be simplified to `new HashMap<>();`

File: src/main/java/com/google/sps/servlets/RecommendServlet.java [hunk 4]
```diff
@@ -37,10 +37,25 @@
...
+    List<String> listStoreIds = new ArrayList<>(); // todo: get this list from the places api
     List<Drink> drinkList = new ArrayList<>();
-    RecommendationEngine.getBestShop(listStoreIds, beverage, drinkList);
+
+    // adds all drinks from the storeid's we want into a list
+    for (String storeId : listStoreIds) {
+      for (Drink drink : DrinkDAO.getDrinksByStore(storeId)) {
```
Comment (albert-g3) [inline src/main/java/com/google/sps/servlets/RecommendServlet.java:49]: getDrinksByStore() returns a list? you can use addAll() method from list

File: src/main/java/com/google/sps/servlets/RecommendServlet.java [hunk 5]
```diff
@@ -37,10 +37,25 @@
...
+        drinkList.add(drink);
+      }
+    }
+
+    // puts the beverage into a hashmap which maps to its bestshop storeid
+    for (String beverage : beverages) {
+      beverage = beverage.trim();
```
Comment (albert-g3) [inline src/main/java/com/google/sps/servlets/RecommendServlet.java:56]: not necessary, just put beverage.trim() in put()

**Review by albert-g3 — APPROVED** [review]

**Pull request merged**